tools/db_bench_tool.cc

```diff
@@ -1965,11 +1972,19 @@ class RandomGenerator {
 
   Slice Generate(unsigned int len) {
     assert(len <= data_.size());
-    if (pos_ + len > data_.size()) {
-      pos_ = 0;
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<> dis(0.0, 1.0);
+    auto prob = dis(gen);
+    if (prob < FLAGS_same_value_rate) {
```

Comment:
For a bernoulli distribution, I believe you can use stuff in random.h like below. RocksDB already wrote its own data structure more often than you thought. 

```
  // Randomly returns true ~"1/n" of the time, and false otherwise.
  // REQUIRES: n > 0
  bool OneIn(int n) { return Uniform(n) == 0; }
 ```

Reply:
Ignore my previous comment - now I understand what is this for. Is it possible to use below from Random.h instead? Maybe also change the name to something like "repeat_value_percentage"? 

```
  bool PercentTrue(int percentage) {
    return static_cast<int>(Uniform(100)) < percentage;
  }
 ```